source/server/connection_handler_impl.h

```diff
@@ -130,6 +128,38 @@ class ConnectionHandlerImpl : public Network::ConnectionHandler, NonCopyable {
     Stats::TimespanPtr conn_length_;
   };
 
+  /**
+   * Wrapper for an active accepted socket owned by this handler.
+   */
+  struct ActiveSocket : public Network::ListenerFilterManager,
+                        public Network::ListenerFilterCallbacks,
+                        LinkedObject<ActiveSocket>,
+                        public Event::DeferredDeletable {
+    ActiveSocket(ActiveListener& listener, Network::AcceptedSocketPtr&& socket, bool redirected)
+        : listener_(&listener), socket_(std::move(socket)), redirected_(redirected),
+          iter_(accept_filters_.end()) {}
+    ~ActiveSocket() {
+      ASSERT(iter_ == accept_filters_.end());
+      accept_filters_.clear();
+    }
+
+    // Network::ListenerFilterManager
+    void addAcceptFilter(Network::ListenerFilterPtr&& filter) override {
+      accept_filters_.emplace_back(std::move(filter));
+    }
+
+    // Network::ListenerFilterCallbacks
+    Network::AcceptedSocket& socket() override { return *socket_.get(); }
+    Event::Dispatcher& dispatcher() override { return listener_->parent_.dispatcher_; }
+    void continueFilterChain(bool success) override;
+
+    ActiveListener* listener_;
```

Comment:
nit: `ActiveListener&`

Reply:
Fixed. Have to get to the habit of avoiding bare pointers...